packages/runtime/test/src/interpreter/opcode-list.ts

```diff
@@ -1988,12 +1992,87 @@ describe("Teal Opcodes", function () {
+        op.execute(stack);
+        assert.equal(1, stack.length());
+        assert.equal(BigInt(TXN_OBJ.apas[0]), stack.pop());
+
+        // should push Accounts[1] to stack
+        op = new Txn(["Assets", "1"], 1, interpreter);
+        op.execute(stack);
+        assert.equal(1, stack.length());
+        assert.equal(BigInt(TXN_OBJ.apas[1]), stack.pop());
+
+        // index 10 should be out_of_bound
+        op = new Txn(["Assets", "10"], 1, interpreter);
+        expectRuntimeError(
+          () => op.execute(stack),
+          RUNTIME_ERRORS.TEAL.INDEX_OUT_OF_BOUND
+        );
+
+        op = new Txn(["NumAssets"], 1, interpreter);
+        op.execute(stack);
+        assert.equal(BigInt(TXN_OBJ.apas.length), stack.pop());
+      });
+
+      it("should push value from foreign applications array and push NumApplications", function () {
+        // special case: Txn.Applications[0] represents current_applications_id
+        let op = new Txn(["Applications", "0"], 1, interpreter);
+        op.execute(stack);
+
+        assert.equal(1, stack.length());
+        assert.equal(BigInt(TXN_OBJ.apid), stack.pop());
+
+        // Txn.Applications[1] should push "1st" app_id from foreign Apps (Txn.ForeignApps[0])
+        op = new Txn(["Applications", "1"], 1, interpreter);
+        op.execute(stack);
+
+        assert.equal(1, stack.length());
+        assert.equal(BigInt(TXN_OBJ.apfa[0]), stack.pop());
+
+        // index 10 should be out_of_bound
+        op = new Txn(["Applications", "10"], 1, interpreter);
+        expectRuntimeError(
+          () => op.execute(stack),
+          RUNTIME_ERRORS.TEAL.INDEX_OUT_OF_BOUND
+        );
+
+        op = new Txn(["NumApplications"], 1, interpreter);
+        op.execute(stack);
+        assert.equal(BigInt(TXN_OBJ.apfa.length), stack.pop());
+      });
+
+      it("should push local, global uint and byte slices from state schema to stack", function () {
+        let op = new Txn(["GlobalNumUint"], 1, interpreter);
+        op.execute(stack);
+        assert.equal(1, stack.length());
+        assert.equal(BigInt(TXN_OBJ.apgs.nui), stack.pop());
+
+        op = new Txn(["GlobalNumByteSlice"], 1, interpreter);
+        op.execute(stack);
+        assert.equal(1, stack.length());
+        assert.equal(BigInt(TXN_OBJ.apgs.nbs), stack.pop());
+
+        op = new Txn(["LocalNumUint"], 1, interpreter);
+        op.execute(stack);
+        assert.equal(1, stack.length());
+        assert.equal(BigInt(TXN_OBJ.apls.nui), stack.pop());
+
+        op = new Txn(["LocalNumByteSlice"], 1, interpreter);
+        op.execute(stack);
+        assert.equal(1, stack.length());
+        assert.equal(BigInt(TXN_OBJ.apls.nbs), stack.pop());
+      });
     });
 
     describe("Gtxn", function () {
       before(function () {
         const tx = interpreter.runtime.ctx.tx;
-        const tx2 = { ...tx, fee: 2222 };
+        const tx2 = { ...tx, fee: 2222, apas: [3033, 4044], apfa: [5005, 6006, 7077] };
```

Comment:
could you please add here what are these values ?

Reply:
yes